air/src/execution_step/execution_context/cid_state.rs

```diff
@@ -138,15 +138,27 @@ impl ExecutionCidState {
             .ok_or_else(|| UncatchableError::ValueForCidNotFound("service result aggregate", cid.clone().into()))
     }
 
-    pub(crate) fn resolve_service_value(
+    pub(crate) fn resolve_service_info(
         &self,
         service_result_agg_cid: &CID<ServiceResultCidAggregate>,
-    ) -> Result<Rc<JValue>, UncatchableError> {
+    ) -> Result<ResolvedServiceInfo, UncatchableError> {
         let service_result_aggregate = self.get_service_result_agg_by_cid(service_result_agg_cid)?;
-        self.get_value_by_cid(&service_result_aggregate.value_cid)
+        let value = self.get_value_by_cid(&service_result_aggregate.value_cid)?;
+        let tetraplet = self.get_tetraplet_by_cid(&service_result_aggregate.tetraplet_cid)?;
```

Comment:
super nit: let's add a new line after this line to split the block variables creation and resulting value creation